Update .NET Core index page (#16680)
dotnet/docs: docs/core/index.md

> ---

Reviewer: update ms.date on line 5

> # .NET Core guide

Reviewer: we're going away from the term guides. Should we start with this PR?

```suggestion
# .NET Core documentation
```

> [.NET Core](about.md) is an [open-source](https://github.com/dotnet/runtime/blob/master/LICENSE.TXT), general-purpose development platform. You can create .NET Core apps for Windows, macOS, and Linux, for x64, x86, ARM32 and ARM64 processors, with [C#](/dotnet/csharp/), [F#](/dotnet/fsharp/) and [VB](/dotnet/visual-basic/) languages. Frameworks and APIs are provided for [cloud](/aspnet/core/), [IoT](/archive/msdn-magazine/2019/august/net-core-cross-platform-iot-programming-with-net-core-3-0), [client UI](/dotnet/desktop-wpf/overview/index), and [machine learning](/dotnet/machine-learning/).

Reviewer: relative links are validated at build time and work better for VS offline books 
do we want to link to Spark as well?

```suggestion
[.NET Core](about.md) is an [open-source](https://github.com/dotnet/runtime/blob/master/LICENSE.TXT), general-purpose development platform. You can create .NET Core apps for Windows, macOS, and Linux, for x64, x86, ARM32 and ARM64 processors, with [C#](../csharp/index.md), [F#](../fsharp/index.md) and [VB](../visual-basic/index.md) languages. Frameworks and APIs are provided for [cloud](/aspnet/core/), [IoT](https://docs.microsoft.com/archive/msdn-magazine/2019/august/net-core-cross-platform-iot-programming-with-net-core-3-0), [client UI](../desktop-wpf/overview/index.md), and [machine learning](../machine-learning/index.md).
```

You can [download the .NET Core SDK](https://dotnet.microsoft.com/download) to try .NET Core on your machine. The latest version is [.NET Core 3.1](https://devblogs.microsoft.com/dotnet/announcing-net-core-3-1/).

See [About .NET Core](about.md) to learn more about .NET Core, and  [What's new in .NET Core 3.1](./whats-new/dotnet-core-3-1.md) to learn about the latest release.

Check out [.NET Core Tutorials](tutorials/index.md) to learn how to create a simple .NET Core application. It only takes a few minutes to get your first app up and running. If you want to try .NET Core in your browser, look at the [Numbers in C#](../csharp/tutorials/intro-to-csharp/numbers-in-csharp.yml) online tutorial.

## Download .NET Core

You can get .NET Core in the following ways:

> * [Installers for Windows and macOS](https://dotnet.microsoft.com/download)

Reviewer: we have articles specific for installation and the different methods. Should we link to that instead?

e.g. https://docs.microsoft.com/en-us/dotnet/core/install/sdk

* [Linux packages](https://docs.microsoft.com/dotnet/core/install/linux-package-managers)
* [Docker containers](https://hub.docker.com/_/microsoft-dotnet-core/)
* [Zips and tar balls](https://dotnet.microsoft.com/download/dotnet-core/3.1)
* [Install scripts](https://dotnet.microsoft.com/download/dotnet-core/scripts)
* [Release notes](https://github.com/dotnet/core/tree/master/release-notes)

## Create your first application

> After installing the .NET Core SDK, open a command prompt. Use the following commands to create and run an application:

Reviewer: why remove the C# from this?

Hello World!
```

> ## Participate

Reviewer: Would contribute read better?

.NET Core is an open platform. Everyone is welcome to participate.

> * Product issues and questions should be filed as issues at [dotnet/core](https://github.com/dotnet/core/issues)

Reviewer: We typically send people for product issues to https://developercommunity.visualstudio.com/spaces/61/index.html given that product issues can belong to different repos (runtime, sdk, etc.). Wouldn't that be better?

```suggestion
* Product issues and questions should be filed as issues at [dotnet/core](https://github.com/dotnet/core/issues).
```

> * Contributions should be made on one of the project repos, such as [dotnet/runtime](https://github.com/dotnet/runtime), [dotnet/sdk](https://github.com/dotnet/sdk), [dotnet/rosyln](https://github.com/dotnet/roslyn), or [aspnetcore](https://github.com/dotnet/aspnetcore)

Reviewer: Should we link here too: https://github.com/dotnet/core/blob/master/Documentation/core-repos.md?

```suggestion
* Product contributions should be made on one of the project repositories, such as [dotnet/runtime](https://github.com/dotnet/runtime), [dotnet/sdk](https://github.com/dotnet/sdk), [dotnet/rosyln](https://github.com/dotnet/roslyn), or [aspnetcore](https://github.com/dotnet/aspnetcore).
```

> Thanks! .NET Core was made with love.

Reviewer: This reads very blog post-like, which is unusual for docs.
